server/datastore/mysql/software_installers.go

```diff
@@ -2316,3 +2316,42 @@ WHERE
 
 	return res, nil
 }
+
+func (ds *Datastore) GetSoftwareInstallerByHash(ctx context.Context, sha256 string) (map[uint]*fleet.ExistingSoftwareInstaller, error) {
+	stmt := `
+SELECT 
+	si.id AS installer_id,
```

Comment:
Do we need title ID here?

Reply:
@iansltx I don't think so, because the title ID is found later on in the flow and as long as these other fields match up, that downstream machinery will find the right title ID